src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -357,14 +367,29 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    if (is.null(outputFile))
       outputFile <- .rs.withChangedExtension(inputFile, ext = ".nb.html")
 
-   # TODO: pass encoding from frontend
-   encoding <- "UTF-8"
+   # attempt to get encoding from source database (note: this will only
+   # succeed for files already open in the IDE, but since this operation
+   # is normally called when attempting to preview / create a notebook on
+   # save we generally expect the document to be available)
+   properties <- .rs.getSourceDocumentProperties(inputFile)
+   
+   # attempt to read and re-encode the file to UTF-8 if it's specified
+   # in the system encoding
+   if (!identical(properties$encoding, "UTF-8")) {
+      contents <- readLines(inputFile)
+      utf8 <- tryCatch(
+         iconv(contents, from = properties$encoding, to = "UTF-8"),
+         error = function(e) contents
```

Comment:
Actually though, now that I look at that code in rmarkdown, it looks like it's doing pretty much exactly what you are doing here -- passing the supplied encoding to `iconv` to get at UTF-8 encoded copy of the file. (see https://github.com/rstudio/rmarkdown/blob/2158b9d6b00f2e0cfd67e0218dfb2f8e8548ad35/R/render.R#L177-L178). Can we just delegate this whole thing to the rmarkdown package?